creusot/src/backend/dependency.rs

```diff
@@ -57,9 +58,13 @@ impl<'tcx> Dependency<'tcx> {
                 TyKind::Adt(def, _) if !def.is_box() => {
                     Some(item_symb(tcx, def.did(), rustc_hir::def::Namespace::TypeNS))
                 }
-                TyKind::Alias(_, aty) => {
-                    Some(Symbol::intern(&type_name(tcx.item_name(aty.def_id).as_str())))
-                }
+                TyKind::Alias(AliasTyKind::Opaque, aty) => Some(Symbol::intern(&format!(
+                    "opaque{}",
+                    tcx.def_path(aty.def_id).data.last().unwrap().disambiguator
+                ))),
+                TyKind::Alias(_, aty) => Some(Symbol::intern(&type_name(
+                    tcx.opt_item_name(aty.def_id).unwrap_or(Symbol::intern("synthetic")).as_str(),
```

Comment:
I'm not sure _how_ but I'm getting these new "synthetic" types in the code I'm looking at which I don't yet know how to handle cleanly here.

Reply:
Do you have a repro? If possible, I'd like to have it in our tests.